test/katex-spec.js

```diff
@@ -1042,6 +1042,16 @@ describe("A left/right parser", function() {
         var normalEmpty = "\\Bigl .";
         expect(normalEmpty).toParse();
     });
+
+    it("should handle \\middle", function() {
+        var normalMiddle = "\\left( \\dfrac{x}{y} \\middle| \\dfrac{y}{z} \\right)";
```

Comment:
Could you try putting more than one `\middle` in here? Also, maybe test `\middle` with nested `\left\right`s?